### PR TITLE
OpenSora: add multi-resolution buckets support

### PR DESCRIPTION
**Pull request by hadipash**
Add multi-resolution buckets support

### CODE REVIEW

**Comment by SamitHuang** [comment]
has the  small bugs of inference fixed? (e.g. pass empty reference_path from CLI)

**Comment by hadipash** [comment]
> has the small bugs of inference fixed? (e.g. pass empty reference_path from CLI)

No, let me add the fixes to this PR.

File: examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py [hunk 1]
```diff
@@ -32,6 +34,22 @@ def create_infer_transforms(target_size: Tuple[int, int], interpolation=Inter.BI
...
+class BucketResizeCrop:
+    def __init__(self, buckets: Bucket):
+        self._transforms = {}  # is this reasonable? There are 350+ buckets
+        for name, lengths in buckets.ar_criteria.items():
+            self._transforms[name] = {}
+            for length, ars in lengths.items():
+                self._transforms[name][str(length)] = {}
+                for ar, hw in ars.items():
+                    self._transforms[name][str(length)][ar] = Compose(
+                        [Resize(min(hw), interpolation=Inter.BILINEAR), CenterCrop(hw)]
```
Comment (SamitHuang) [inline examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py:46]: Need to improve the transformation considering different AR.

src: 1024, 1024
tar: (576, 1024)

current:
resize: (576, 576) 
center crop:  (576, 1024)   black region will be filled on left and right side

**Comment by zhtmike** [comment]
maybe better to add a configure file for t2v for sora1.1, allow use easy to use

File: examples/opensora_hpcai/configs/opensora-v1-1/inference/sample_t2v.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+batch_size: 1
+
+# sampling
+sampling_steps: 100
+guidance_scale: 7.0
+guidance_channels: 3
+seed: 42
+ddim_sampling: False
+
+loop: 2
```
Comment (zhtmike) [inline examples/opensora_hpcai/configs/opensora-v1-1/inference/sample_t2v.yaml:21]: I think loop=1 may be bettter for a more standard i2v task unless users know that they want to loop from prveious frame

Comment (hadipash) [reply examples/opensora_hpcai/configs/opensora-v1-1/inference/sample_t2v.yaml]: done

**Review by SamitHuang — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

File: examples/opensora_hpcai/opensora/models/stdit/stdit2.py [hunk 3]
```diff
@@ -238,16 +237,12 @@ def __init__(
             self.x_embedder = PatchEmbed3D(patch_size, in_channels, hidden_size)
         elif patchify_conv3d_replace == "linear":
             assert patch_size[0] == 1 and patch_size[1] == patch_size[2]
-            # assert input_size[1] == input_size[2]
             print("Replace conv3d patchify with linear layer")
-            self.x_embedder = LinearPatchEmbed(input_size[1], patch_size[1], in_channels, hidden_size, bias=True)
+            self.x_embedder = LinearPatchEmbed(in_channels, hidden_size, bias=True)
```
Comment (zhtmike) [inline examples/opensora_hpcai/opensora/models/stdit/stdit2.py:241]: patch_size[1] should not be deleted.

**Pull request merged**